accepted/0000-npm-diff.md

```diff
@@ -0,0 +1,60 @@
+# npm diff
+
+## Summary
+
+Add a new command that enables diff workflows (similar to `git diff`) for packages published in the registry.
+
+## Motivation
+
+- Complements `npm audit` and `npm outdated` workflows by providing insight on what changed across different package versions
+- Enables package authors to diff packlist-tracked-only file changes prior to publishing a new version of a package
+
+## Detailed Explanation
+
+Introduce a new `npm diff` command that will fetch tarball contents for two different versions of a module and print that output in an usable format to users.
+
+### npm diff `<spec>`
+
+Using a single `<spec>` allows users to retrieve diff output between an existing valid version of a package with the same **name** found in the actual dependency tree and the exact `<spec>` match from the registry.
+
+<img alt="Demo showing npm diff pkg in a repo with an actual tree installed" src="https://ruyadorno.github.io/svg-demos/npm-diff/npm-diff-yargs.svg" />
+
+### npm diff `<spec-a>` `<spec-b>`
+
+Fetches tarball contents of two versions of a package, represented by `<spec-a>` and `<spec-b>` and prints diff output.
+
+<img alt="Demo showing npm diff pkg in a repo with an actual tree installed" src="https://ruyadorno.github.io/svg-demos/npm-diff/npm-diff-simple-output.svg" />
+
+### npm diff --changelog
```

Comment:
It would be wonderful to see the ability to pick a single file from the `npm diff` output. Perhaps this could just be a shortcut built on top of that?

This also leads me down the path of wondering if it would be possible to include CHANGELOG files in the npmjs.com UI.

Reply:
oh yeah, a proper changelog section in the website UI would be neat, specially one that does fallback to looking up GitHub releases 🤩 

now back to your original point, I think it makes a lot of sense to have a file-filter argument, as I commented during the call it would enable us to do neat things in companion of the `--name-only` flag, workflows such as: `npm diff --name-only | grep index | xargs git diff` or have an interactive workflow that allows you to select from the list of files: `npm diff --name-only | npx ipt | xargs git diff`